src/dtgtk/thumbtable.c

```diff
@@ -542,25 +593,12 @@ static int _thumbs_load_needed(dt_thumbtable_t *table)
     {
       if(posy < table->view_height) // we don't load invisible thumbs
       {
-        dt_thumbnail_t *thumb = dt_thumbnail_new
-          (table->thumb_size,
-           table->thumb_size,
-           IMG_TO_FIT,
-           sqlite3_column_int(stmt, 1),
-           sqlite3_column_int(stmt, 0), table->overlays,
-           DT_THUMBNAIL_CONTAINER_LIGHTTABLE, table->show_tooltips);
-
-        if(table->mode == DT_THUMBTABLE_MODE_FILMSTRIP)
-        {
-          thumb->single_click = TRUE;
-          thumb->sel_mode = DT_THUMBNAIL_SEL_MODE_MOD_ONLY;
-        }
-        thumb->x = posx;
-        thumb->y = posy;
-        table->list = g_list_prepend(table->list, thumb);
-        gtk_widget_set_margin_start(thumb->w_image_box, old_margin_start);
-        gtk_widget_set_margin_top(thumb->w_image_box, old_margin_top);
-        gtk_layout_put(GTK_LAYOUT(table->widget), thumb->w_main, posx, posy);
+        _thumb_move_or_create(table,
```

Comment:
For clarity please use intermediate variable to name the parameter:

```
const dt_imgid_timgid = sqlite3_column_int(stmt, 1);
const int rowid =  sqlite3_column_int(stmt, 0);
```

And use them below.